internal/source/source.go

```diff
@@ -0,0 +1,67 @@
+package source
+
+import (
+	"context"
+	"errors"
+	"io/fs"
+	"sync"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+type Source interface {
+	SourceContent(context.Context, *rukpakv1alpha1.Bundle) (*Result, error)
+}
+
+type Result struct {
```

Comment:
How do we feel about SourceResult (or anything slightly more explicit that could potentially improve readability)?

Reply:
Also: nit: godoc to implement readability?